---
title: "mitovizR"
author: 
  name: "Roberto Preste" 
  email: "roberto.preste@gmail.com"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{mitovizR}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction  

The mitochondrial genome is the circular genomic DNA contained in mitochondria, organelles located in the cytoplasm of eukaryotic cells.  
In humans, the mitochondrial genome is 16569 bp long, and contains 37 genes, 13 of which are protein-coding genes. Most of the others are tRNA-coding genes, while 2 are rRNA-coding.  

The recent advances in high-throughput DNA sequencing allow a thorough exploration and analysis of mitochondrial variants, which are positions in which nucleotides differ from the reference human mitochondrial genome.  

While there are lots of tools available to analyse human mitochondrial genomes, visualisation of human mitochondrial variants usually relies on researchers' knowledge of general programming languages with visualisation capabilities. In order to fill this gap, I developed the `mitovizR` package, which offers several visualisation options right out of the box, with a simple user interface using the R programming language.  

## Installation  

`mitovizR` can be installed using `devtools::install_github()`, as follows:  

```{r, eval=FALSE}
library(devtools)
install_github("robertopreste/mitovizR")
```

This will download and install the latest development version of `mitovizR` right from its GitHub repository. This means that some of its features can still be in beta, and might not function as expected.  



